os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -746,6 +746,13 @@ int binary_manager_read_header(int type, char *devpath, void *header_data, bool
 			bin_size = ((common_binary_header_t *)header_data)->bin_size;
 			crc_hash = ((common_binary_header_t *)header_data)->crc_hash;
 		}
+		struct mallinfo mem;
+#ifdef CONFIG_CAN_PASS_STRUCTS
+		mem = kmm_mallinfo();
+#else
+		(void)kmm_mallinfo(&mem);
+#endif
+		crc_bufsize = crc_bufsize < mem.mxordblk ? crc_bufsize : (mem.mxordblk / 2);
```

Comment:
This looks too dangerous. Even if crc_bufsize == mem.mxordblk - 1, crc_bufsize will be used. Then other thread will be failed to allocate. How about checking half of mem.mxordblk?
```
crc_bufsize = crc_bufsize < (mem.mxordblk / 2) ? crc_bufsize : (mem.mxordblk / 2);
```